content/courses/native-onchain-development/program-derived-addresses.md

```diff
@@ -104,25 +103,16 @@ function. If the output of `create_program_address` is not a valid PDA, then the
 
 ```rust
 pub fn try_find_program_address(seeds: &[&[u8]], program_id: &Pubkey) -> Option<(Pubkey, u8)> {
-
-    let mut bump_seed = [std::u8::MAX];
-    for _ in 0..std::u8::MAX {
-        {
-            let mut seeds_with_bump = seeds.to_vec();
-            seeds_with_bump.push(&bump_seed);
-            match Self::create_program_address(&seeds_with_bump, program_id) {
-                Ok(address) => return Some((address, bump_seed[0])),
-                Err(PubkeyError::InvalidSeeds) => (),
-                _ => break,
-            }
+    for bump in (0..=std::u8::MAX).rev() {
```

Comment:
What does the `=` do here?

Reply:
range inclusion of all possible u8-bit values ie not excludes the end point.